test/src/main/java/org/apache/accumulo/test/TransportCachingIT.java

```diff
@@ -49,122 +45,78 @@
-        } catch (TTransportException e) {
-          log.warn("Failed to obtain transport to {}", servers);
-        }
-      }
+      TTransport first = getAnyTransport(servers, pool, true);
 
       assertNotNull(first);
       // Return it to unreserve it
       pool.returnTransport(first);
 
-      TTransport second = null;
-      while (second == null) {
-        try {
-          // Get a cached transport (should be the first)
-          second = pool.getAnyTransport(servers, true).getSecond();
-        } catch (TTransportException e) {
-          log.warn("Failed obtain 2nd transport to {}", servers);
-        }
-      }
+      TTransport second = getAnyTransport(servers, pool, true);
 
       // We should get the same transport
       assertSame("Expected the first and second to be the same instance", first, second);
-      // Return the 2nd
       pool.returnTransport(second);
 
-      TTransport third = null;
-      while (third == null) {
-        try {
-          // Get a non-cached transport
-          third = pool.getAnyTransport(servers, false).getSecond();
-        } catch (TTransportException e) {
-          log.warn("Failed obtain 3rd transport to {}", servers);
-        }
-      }
-
+      // Ensure does not get cached connection just returned
+      TTransport third = getAnyTransport(servers, pool, false);
       assertNotSame("Expected second and third transport to be different instances", second, third);
-      pool.returnTransport(third);
 
-      // ensure the LIFO scheme with a fourth and fifth entry
-      TTransport fourth = null;
-      while (fourth == null) {
-        try {
-          // Get a non-cached transport
-          fourth = pool.getAnyTransport(servers, false).getSecond();
-        } catch (TTransportException e) {
-          log.warn("Failed obtain 4th transport to {}", servers);
-        }
-      }
+      TTransport fourth = getAnyTransport(servers, pool, false);
+      assertNotSame("Expected third and fourth transport to be different instances", third, fourth);
+
+      pool.returnTransport(third);
       pool.returnTransport(fourth);
-      TTransport fifth = null;
-      while (fifth == null) {
-        try {
-          // Get a cached transport
-          fifth = pool.getAnyTransport(servers, true).getSecond();
-        } catch (TTransportException e) {
-          log.warn("Failed obtain 5th transport to {}", servers);
-        }
-      }
+
+      // The following three asserts ensure the per server queue is LIFO
+      TTransport fifth = getAnyTransport(servers, pool, true);
       assertSame("Expected fourth and fifth transport to be the same instance", fourth, fifth);
+
+      TTransport sixth = getAnyTransport(servers, pool, true);
+      assertSame("Expected third and sixth transport to be the same instance", third, sixth);
+
+      TTransport seventh = getAnyTransport(servers, pool, true);
+      assertSame("Expected third and sixth transport to be the same instance", second, seventh);
```

Comment:
Comment is incorrect:
```suggestion
      assertSame("Expected second and seventh transport to be the same instance", second, seventh);
```